src/components/search/HPSearchBar.tsx

```diff
@@ -7,8 +7,9 @@ import { identity } from "../../utils/identity";
 //TODO: Note HPSearchBar is sharing its searchBarValue between screens.
 export const HPSearchBar = ({searchBarValue: value = "", placeholder, suggestions, renderSuggestion, onSuggestionSelected, onSearchBarValueChange}) => {
     const getSuggestionValue = identity
+
     const onChange = ({target: { value }}:React.ChangeEvent<HTMLInputElement>) => 
-        onSearchBarValueChange(value)
+        (value as any !== 0) && onSearchBarValueChange(value || '')
```

Comment:
what's the point of doing this check here if the same check is being done in the `onSearchBarValueChange` function?

Can we also leave a comment on exactly what the bug is if we don't check for 0? I keep forgetting :(

Reply:
killed the outside one (handle inside so no caller ever needs to)
updated condition with self explanatory name